src/Propel/Runtime/ActiveQuery/Lock.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types = 1);
+
+namespace Propel\Runtime\ActiveQuery;
+
+/**
+ * Class represents a query lock
+ *
+ * @author Tomasz Wójcik <tomasz.prgtw.wojcik@gmail.com>
+ */
+class Lock
+{
+    public const SHARED = 'SHARED';
+
+    public const EXCLUSIVE = 'EXCLUSIVE';
+
+    /**
+     * Lock type, either shared or exclusive
+     *
+     * @see self::SHARED
+     * @see self::EXCLUSIVE
+     *
+     * @var string
+     */
+    private $type;
```

Comment:
Not sure what the rest of the system is using
But can we also use protected visibility here?
If anyone needs to extend this, this would make it much harder in that rare case - for no real benefit at this point.

Reply:
sure, I've changed to `protected` as requested, makes more sense if there will be children classes in the future